### PR TITLE
PYTHON-4309 OIDC Atlas Testing Updates

### PR DESCRIPTION
**Pull request by blink1073**
Requires https://github.com/mongodb-labs/drivers-evergreen-tools/pull/426

### CODE REVIEW

**Comment by blink1073** [comment]
Windows failure is expected pending: https://jira.mongodb.org/browse/DEVPROD-5919

File: .evergreen/run-mongodb-oidc-test.sh [hunk 1]
```diff
@@ -13,52 +13,11 @@ if [ $OIDC_ENV == "test" ]; then
...
-        export MONGODB_URI_SINGLE="$OIDC_ATLAS_URI_SINGLE/?authMechanism=MONGODB-OIDC"
-        export MONGODB_URI_MULTI="$OIDC_ATLAS_URI_MULTI/?authMechanism=MONGODB-OIDC"
-        set -x
-    fi
-    export OIDC_TOKEN_FILE="$OIDC_TOKEN_DIR/test_user1"
-    set +x   # turn off xtrace for this portion
-    export OIDC_ADMIN_USER=$OIDC_ATLAS_USER
-    export OIDC_ADMIN_PWD=$OIDC_ATLAS_PASSWORD
-    set -x
+    source ${DRIVERS_TOOLS}/.evergreen/auth_oidc/secrets-export.sh
```
Comment (NoahStapp) [inline .evergreen/run-mongodb-oidc-test.sh:16]: Amazing change, so much cleaner.

File: .evergreen/run-tests.sh [hunk 2]
```diff
@@ -221,7 +221,7 @@ fi
 
 if [ -n "$TEST_AUTH_OIDC" ]; then
     python -m pip install ".[aws]"
-    TEST_ARGS="test/auth_oidc/test_auth_oidc.py"
+    TEST_ARGS="test/auth_oidc/test_auth_oidc.py $TEST_ARGS"
```
Comment (NoahStapp) [inline .evergreen/run-tests.sh:224]: This is prepending `test/auth_oidc/test_auth_oidc.py` to the existing `TEST_ARGS`?

File: test/auth_oidc/test_auth_oidc.py [hunk 3]
```diff
@@ -56,7 +58,7 @@ class OIDCTestBase(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
         cls.uri_single = os.environ["MONGODB_URI_SINGLE"]
-        cls.uri_multiple = os.environ["MONGODB_URI_MULTI"]
+        cls.uri_multiple = os.environ.get("MONGODB_URI_MULTI")
```
Comment (NoahStapp) [inline test/auth_oidc/test_auth_oidc.py:61]: Should these all be `os.environ.get` for consistency?

**Review by NoahStapp — CHANGES_REQUESTED** [review]
A few clarifying questions, otherwise looks great!

File: .evergreen/run-tests.sh (hunk 2 above)
Comment (blink1073) [reply .evergreen/run-tests.sh:224]: Yes

File: test/auth_oidc/test_auth_oidc.py (hunk 3 above)
Comment (blink1073) [reply test/auth_oidc/test_auth_oidc.py:61]: No, the others have to be there, we want it to error out.

**Review by NoahStapp — APPROVED** [review]

**Pull request merged**